x-pack/plugins/rule_registry/common/field_map/runtime_type_from_fieldmap.ts

```diff
@@ -6,22 +6,56 @@
  */
 import { Optional } from 'utility-types';
 import { mapValues, pickBy } from 'lodash';
+import { either } from 'fp-ts/lib/Either';
 import * as t from 'io-ts';
 import { FieldMap } from './types';
 
+const NumberFromString = new t.Type(
+  'NumberFromString',
+  (u): u is number => typeof u === 'number',
+  (u, c) =>
+    either.chain(t.string.validate(u, c), (s) => {
+      const d = Number(s);
+      return isNaN(d) ? t.failure(u, c) : t.success(d);
+    }),
+  (a) => a
+);
+
+const BooleanFromString = new t.Type(
+  'BooleanFromString',
+  (u): u is boolean => typeof u === 'boolean',
+  (u, c) =>
+    either.chain(t.string.validate(u, c), (s) => {
+      switch (s.toLowerCase().trim()) {
+        case '1':
+        case 'true':
+        case 'yes':
+          return t.success(true);
+        case '0':
+        case 'false':
+        case 'no':
+        case null:
+          return t.success(false);
+        default:
+          return t.failure(u, c);
+      }
+    }),
+  (a) => a
+);
+
 const esFieldTypeMap = {
   keyword: t.string,
   text: t.string,
   date: t.string,
-  boolean: t.boolean,
-  byte: t.number,
-  long: t.number,
-  integer: t.number,
-  short: t.number,
-  double: t.number,
-  float: t.number,
-  scaled_float: t.number,
-  unsigned_long: t.number,
+  boolean: t.union([t.number, BooleanFromString]),
```

Comment:
Would using `t.boolean` instead of `t.number`, e.g:

```
t.union([t.boolean, BooleanFromString])
```

make sense here?

Reply:
bad copy pasta, thank you!